src/resmom/start_exec.c

```diff
@@ -2182,7 +2192,8 @@ receive_string_data(int downfds, int  upfds)
 		return (NULL);
 	}
 	/* send back as an acknowledgement that MOM got it */
-	if (write_pipe_data(upfds, r_buf, r_size) != 0) {
+	ack_size = r_size;
+	if (write_pipe_data(upfds, &ack_size, sizeof(size_t)) != 0) {
 		snprintf(log_buffer, sizeof(log_buffer), "write of %lu to pipe failed", r_size);
```

Comment:
How about... write of length 1234 bytes to pipe failed

Reply:
Ok, this one too. Fixed.